plugins/follow_me/follow_me_impl.cpp

```diff
@@ -0,0 +1,213 @@
+
+    _parent->register_mavlink_message_handler(
+        MAVLINK_MSG_ID_HEARTBEAT,
+        std::bind(&FollowMeImpl::process_heartbeat, this, _1), static_cast<void *>(this));
+}
+
+void FollowMeImpl::deinit()
+{
+    _parent->unregister_all_mavlink_message_handlers(this);
+}
+
+void FollowMeImpl::timeout_occurred()
+{
+    // FIXME: This is HARD-coded as of now.
+    // Ideally GPS info should come from a platform-specific framwork.
+
+    // update current location coordinates
+    _motion_report.lat_int += 10;
+    _motion_report.lon_int += 5;
+    _motion_report.alt += 10.0f;
+
+    _estimatation_capabilities |= (1 << static_cast<int>(FollowMe::ESTCapabilities::POS));
+
+    // update current eph & epv
+    _motion_report.pos_std_dev[0] += _motion_report.pos_std_dev[1] += 0.05f;
+    _motion_report.pos_std_dev[2] += 0.05f;
+
+    // calculate z velocity if it's available
+    _motion_report.vz += 0.5f;
+
+    // calculate x,y velocity if it's
+    _motion_report.vx += 0.05f;
+    _motion_report.vy += 0.04f;
+
+    _estimatation_capabilities |= (1 << static_cast<int>(FollowMe::ESTCapabilities::VEL));
+
+    send_gcs_motion_report();
+}
+
+/**
+ * @brief FollowMeImpl::set_motion_report
+ * @param mr
+ */
+void FollowMeImpl::set_motion_report(const FollowMe::MotionReport &mr)
+{
+    _motion_report = mr;
+}
+
+/**
+ * @brief FollowMeImpl::start
+ * @param mr
+ * @return
+ */
+FollowMe::Result FollowMeImpl::start(const FollowMe::MotionReport &mr)
+{
+    // save motion report provided by application
+    _motion_report = mr;
+    return start();
+}
+
+/**
+ * @brief FollowMeImpl::start
+ * @return
+ */
+FollowMe::Result FollowMeImpl::start()
+{
+    _parent->register_timeout_handler(std::bind(&FollowMeImpl::timeout_occurred, this), 1.0,
+                                      &_timeout_cookie);
+    // Note: the safety flag is not needed in future versions of the PX4 Firmware
+    //       but want to be rather safe than sorry.
+    uint8_t flag_safety_armed = _parent->is_armed() ? MAV_MODE_FLAG_SAFETY_ARMED : 0;
+
+    uint8_t mode = MAV_MODE_FLAG_CUSTOM_MODE_ENABLED | flag_safety_armed;
+    uint8_t custom_mode = px4::PX4_CUSTOM_MAIN_MODE_AUTO;
+    uint8_t custom_sub_mode = px4::PX4_CUSTOM_SUB_MODE_AUTO_FOLLOW_TARGET;
+
+    return get_result_from_mavcmd_result(
+               _parent->send_command_with_ack(
+                   MAV_CMD_DO_SET_MODE,
+                   MavlinkCommands::Params {float(mode),
+                                            float(custom_mode),
+                                            float(custom_sub_mode),
+                                            NAN, NAN, NAN, NAN},
+                   MavlinkCommands::DEFAULT_COMPONENT_ID_AUTOPILOT));
+}
+
+/**
+ * @brief FollowMeImpl::send_gcs_motion_report
+ */
+void FollowMeImpl::send_gcs_motion_report()
+{
+    using namespace std::chrono;
+
+    auto end = system_clock::now();
+    auto elapsed_seconds = system_clock::to_time_t(end) - system_clock::to_time_t(_start);
```

Comment:
For everything with time please re-use the time functionality from `global_include.h`. This way we can also mock time for testing, and keep it consistent.

Reply:
Done.